components/action-mgt/org.wso2.carbon.identity.action.execution/src/main/java/org/wso2/carbon/identity/action/execution/ActionExecutorServiceImpl.java

```diff
@@ -85,26 +85,77 @@ public boolean isExecutionEnabled(ActionType actionType) {
         } catch (ActionExecutionRuntimeException e) {
             // todo: add to diagnostics
             LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
             return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        }
+    }
+
+    /**
+     * Resolve the actions by given the action id list and execute them.
+     *
+     * @param actionType    Action Type.
+     * @param actionIds     Lis of action Ids of the actions that need to be executed.
+     * @param eventContext  The event context of the corresponding flow.
+     * @param tenantDomain  Tenant domain.
+     * @return Action execution status.
+     */
+    public ActionExecutionStatus execute(ActionType actionType, String[] actionIds, Map<String, Object> eventContext,
+                                         String tenantDomain) throws ActionExecutionException {
 
+        try {
+            // As of now only one action is allowed.
+            if (actionIds.length != 1) {
+                throw new ActionExecutionException("Only support list with single action Id.");
+            }
+            return execute(getActionByActionId(actionType, actionIds[0], tenantDomain), eventContext);
+        } catch (ActionExecutionRuntimeException e) {
+            // todo: add to diagnostics
+            LOG.debug("Skip executing actions for action type: " + actionType.name(), e);
+            return new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext);
+        }
+    }
+
+    private ActionExecutionStatus execute(Action action, Map<String, Object> eventContext)
+            throws ActionExecutionException {
+
+        ActionType actionType = ActionType.valueOf(action.getType().getActionType());
+        ActionExecutionRequest actionRequest = buildActionExecutionRequest(actionType, eventContext);
+        ActionExecutionResponseProcessor actionExecutionResponseProcessor = getResponseProcessor(actionType);
+
+        return Optional.ofNullable(action)
+                .filter(activeAction -> activeAction.getStatus() == Action.Status.ACTIVE)
+                .map(activeAction -> executeAction(activeAction, actionRequest, eventContext,
+                        actionExecutionResponseProcessor))
+                .orElse(new ActionExecutionStatus(ActionExecutionStatus.Status.FAILED, eventContext));
+    }
+
+    private Action getActionByActionId(ActionType actionType, String actionId, String tenantDomain)
+            throws ActionExecutionRuntimeException {
+
+        try {
+            return ActionExecutionServiceComponentHolder.getInstance().getActionManagementService().getActionByActionId(
+                    Action.ActionTypes.valueOf(actionType.name()).getActionType(), actionId, tenantDomain);
+        } catch (ActionMgtException e) {
+            throw new ActionExecutionRuntimeException("Error occurred while retrieving action by action Id .", e);
```

Comment:
we can cover up this scenario in tests cases. Let's do that

Reply:
Unit tests are added now.